src/main/java/RssReaderAPI/Services/RssFeedsService.java

```diff
@@ -29,147 +26,145 @@ public static RssFeedsService getInstance(){
+        return newsList.subList((int)start, (int)end);
     }
 
-    public String[] getMostUsedWords(long feedId, long rssId) throws FeedNotFoundException {
-        String[] mostUsedWords = new String[5];
-        RssDto rssDto = INSTANCE.getRssFromFeedById(feedId, rssId);
-        try {
-            HashMap<String, Integer> hashMap = new HashMap<>();
-            StringTokenizer tokenizer = new StringTokenizer(rssDto.getDescription(), " ,.:;!?\"\t\n");
-            while (tokenizer.hasMoreTokens()){
-                final String token = tokenizer.nextToken();
-                final Integer value = hashMap.get(token);
+    public RssDto getSingleNews(long feedId, long newsId) throws ResourceNotFoundException, InternalServerError {
+        List<RssDto> newsList= INSTANCE.getAllNews(feedId);
+        if(newsId > newsList.size() - 1) {
+            throw new ResourceNotFoundException("News with id : " + newsId + " not found in feed with id: " + feedId + ". Can't get it!");
+        } else if(newsId < 0) {
+            throw new ResourceNotFoundException("News with id : " + newsId + " not found in feed with id: " + feedId + ". Can't get it!");
+        }
+        return newsList.get((int)newsId);
+    }
 
-                if(value != null){
-                    hashMap.put(token, value + 1);
-                } else {
-                    hashMap.put(token, 1);
-                }
-            }
-            List<Map.Entry<String, Integer>> list = new ArrayList(hashMap.entrySet());
-            Collections.sort(list, new Comparator<Map.Entry<String, Integer>>(){
-                @Override
-                public int compare(Map.Entry<String, Integer> o1, Map.Entry<String, Integer> o2) {
-                    return o2.getValue() - o1.getValue();
-                }
-            });
-            for(int i = 0; i < 5; ++i){
-                mostUsedWords[i] = "Word: " + list.get(i).getKey() + " was used " + list.get(i).getValue().toString() + " times";
+    public String[] getMostUsedWords(long feedId, long rssId, int count) throws ResourceNotFoundException, InternalServerError {
+        RssDto rssDto = INSTANCE.getSingleNews(feedId, rssId);
+        Map<String, Integer> map = new HashMap<>();
+        StringTokenizer tokenizer = new StringTokenizer(rssDto.getDescription(), " ,.:;!?\"\t\n");
+        while (tokenizer.hasMoreTokens()){
+            String token = tokenizer.nextToken();
+            Integer value = map.get(token);
+            if(value != null){
+                map.put(token, value + 1);
+            } else {
+                map.put(token, 1);
             }
-        } catch (Exception e){
-            e.printStackTrace();
+        }
+        List<Map.Entry<String, Integer>> list = new ArrayList(map.entrySet());
+        list.sort((o1, o2) -> o2.getValue() - o1.getValue());
+        String[] mostUsedWords = new String[count];
+        for(int i = 0; i < count; ++i){
+            mostUsedWords[i] = list.get(i).getKey();
```

Comment:
Corner cases are not covered! What if in your list you will have less then 'count' words! -> IndexOutOfBound Exception,
This is unexpected. Even if I requested 5 most used words, the code must return something even if it has not exactly 5